src/include/misc_inline.h

```diff
@@ -315,6 +315,36 @@ __wt_read_shared_double(double *to_read)
     return (*to_read);
 }
 
+/*
+ * __wt_set_shared_maximum64 --
+ *     Given an integer in memory overwrite it with a new value if it is larger.
+ */
+static WT_INLINE void
+__wt_set_shared_maximum64(uint64_t *shared_value, uint64_t new_val)
+{
+    uint64_t current;
+
```

Comment:
Well that is a fairly niche instruction, cool that it exists. I am concerned that we'd be adding another section of ARM specific assembly. ARM assembly requires testing, could at some point change, and is not clear to the reader although given the context of the function it should be obvious as to what it does. 

At this stage the new function is not being called from anywhere in a performance path, in the future it could be. As it stands adding this assembly feels like a case of early optimization with a decent overhead to support that optimization. For that reason I am declining to add this change unless there is a bigger need for it.

Reply:
What do you mean by "could at some point change"? It is part of the armv8.1 ISA specification. That isn't something that just changes (except to make stronger guarantees than before).

I certainly won't require you to use asm here, but I do think it is worth it to make your atomic primitives as efficient as possible on your supported platforms (while still providing the desired guarantees). I recognize that the current use case may not be perf critical, but once the primitive is available it may be used in more places.

Reply:
It may be used in more places but the CAS loop is the standard way of implementing it, until we get gcc support for `LDUMAXAL` we'll leave it as it.